### PR TITLE
URDB v7 response formatted to cmod_utilityrate5

### PR DESCRIPTION
**Pull request by dguittet**
`ElectricityRates_format_as_URDBv7` translates inputs from cmod_utilityrate5 to match as well as possible to [URDB v7 response](https://openei.org/services/doc/rest/util_rates/?version=7)

Fixes issues with `Reopt_size_battery_params` not correctly formatting translating inputs in #333

### CODE REVIEW

File: build_vs2017/ssc_vs2017.vcxproj [hunk 1]
```diff
@@ -44,7 +44,6 @@
     <ClCompile Include="..\ssc\cmod_merchantplant.cpp" />
     <ClCompile Include="..\ssc\cmod_merchantplant_eqns.cpp" />
     <ClCompile Include="..\ssc\cmod_mhk_costs.cpp" />
-    <ClCompile Include="..\ssc\cmod_mhk_eqns.cpp" />
```
Comment (sjanzou) [inline build_vs2017/ssc_vs2017.vcxproj]: build_vs2017 eliminated but cmod_mhk_eqns should remain

File: ssc/cmod_pvsamv1_eqns.cpp [hunk 2]
```diff
@@ -218,79 +219,9 @@ SSCEXPORT void Reopt_size_battery_params(ssc_data_t data) {
...
-        double rate = mat.row(i)[3];
-        var_data rate_data;
-        rate_data.type = SSC_TABLE;
-        rate_data.table.assign("rate", rate);
-        vd_vec.push_back(rate_data);
-        vd_mat.push_back(vd_vec);
-    }
-    reopt_utility.assign("flatdemandstructure", vd_mat);
-    vd_mat.clear();
+    ElectricityRates_format_as_URDBv7(vt);
```
Comment (sjanzou) [inline ssc/cmod_pvsamv1_eqns.cpp]: Nice refactoring

File: ssc/cmod_utilityrate5.cpp [hunk 3]
```diff
@@ -1394,10 +1394,10 @@ class cm_utilityrate5 : public compute_module
 				size_t cnt;
 				ssc_number_t * ts_sr, *ts_br;
 				ts_sr = as_array("ur_ts_sell_rate", &cnt);
-				ts_br = as_array("ur_ts_buy_rate", &cnt);
 				size_t ts_step_per_hour = cnt / 8760;
 				if (ts_step_per_hour < 1 || ts_step_per_hour > 60 || ts_step_per_hour * 8760 != cnt)
 					throw exec_error("utilityrate5", util::format("invalid number of sell rate records (%d): must be an integer multiple of 8760", (int)cnt));
+				ts_br = as_array("ur_ts_buy_rate", &cnt);
```
Comment (sjanzou) [inline ssc/cmod_utilityrate5.cpp]: separated buy rate error?

File: test/ssc_test/cmod_pvsamv1_test.cpp [hunk 4]
```diff
@@ -674,4 +675,25 @@ TEST_F(CMPvsamv1PowerIntegration_cmod_pvsamv1, bifacial_cmod_pvsamv1) {
 		ssc_data_get_number(data, "annual_energy", &annualEnergy);
 		EXPECT_NEAR(annualEnergy, 9141, 1.0) << "Bifacial annual energy from SAM version 2018.11.11 using Phoenix TMY2";
 	}
-}
\ No newline at end of file
+}
```
Comment (sjanzou) [inline test/ssc_test/cmod_pvsamv1_test.cpp]: Thanks for adding

File: ssc/cmod_utilityrate5.cpp (hunk 3 above)
Comment (dguittet) [reply ssc/cmod_utilityrate5.cpp]: I was actually pushed to make this change by a pysam user who had an acceptable sell rate `ur_ts_buy_rate` but no sell rate. 

The original formulation will check the count of the sell rate but then throw an error about the buy rate, so I moved the checks to be appropriate for the error.

File: ssc/cmod_pvsamv1_eqns.cpp (hunk 2 above)
Comment (dguittet) [reply ssc/cmod_pvsamv1_eqns.cpp]: Thanks!

File: build_vs2017/ssc_vs2017.vcxproj (hunk 1 above)
Comment (dguittet) [reply build_vs2017/ssc_vs2017.vcxproj]: That was the bug I'd introduced!

**Pull request merged**